pulp_puppet_plugins/pulp_puppet/plugins/migrations/0005_puppet_module_name_change.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from mongoengine import NotUniqueError
+
+from pulp.server.controllers import repository as repo_controller
+from pulp.server.db import model
+
+from pulp_puppet.plugins.db.models import Module
+
+_log = logging.getLogger('pulp')
+
+
+def migrate(*args, **kwargs):
+    """
+    For each puppet module check and if needed update module name format.
+    """
+    modules = Module.objects(__raw__={'name': {'$regex': '/|-'}})
+    repos_to_rebuild = set()
+    for puppet_unit in modules:
+        try:
+            author, name = puppet_unit['name'].split("-", 1)
+        except ValueError:
+            # This is the forge format, but Puppet still allows it
+            author, name = puppet_unit['name'].split("/", 1)
+        try:
+            puppet_unit.update(set__name=name)
```

Comment:
The mongoengine way to write this is:
```
puppet_unit.name = name
puppet_unit.save()
```

This is better because the way you interact with the object is not mongodb specific with the `set_` call.